pkg/registry/apis/provisioning/resources/folderid.go

```diff
@@ -0,0 +1,76 @@
+package resources
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"path"
+	"regexp"
+	"strings"
+)
+
+var disallowedKubernetesCharacters = regexp.MustCompile(`[^-a-z0-9]`)
+
+// FolderID contains the identifier data for a folder.
+type FolderID struct {
+	// Title is the human-readable name created by a human who wrote it.
+	Title string
+	// KubernetesName represents the name the folder should have, derived from the title.
+	// It contains a suffix calculated from the path of the folder.
+	KubernetesName string
```

Comment:
Should we call this `ID` or `Name`, regardless of K8s or not we should have a unique identifier for the folders.